src/main/java/org/diatliuk/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -93,10 +98,12 @@ public CartItemDto update(Long cartItemId, UpdateCartItemDto updateDto) {
     }
 
     @Override
-    public void deleteById(Long cartItemId) {
-        User authenticatedUser = userService.getAuthenticatedUser();
+    public void deleteById(Authentication authentication, Long cartItemId) {
+        User authenticatedUser = userService.getAuthenticatedUser(authentication);
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .getShoppingCartByUserId(authenticatedUser.getId());
 
-        if (!isUserAbleToModifyItem(authenticatedUser.getId(), cartItemId)) {
+        if (!isUserAbleToModifyItem(shoppingCart, cartItemId)) {
```

Comment:
What is the purpose of this check? It such a situation possible when user can't remove an item from own shopping cart?